src/Symfony/Component/HttpFoundation/IpUtils.php

```diff
@@ -61,26 +66,31 @@ public static function checkIp($requestIp, $ips)
      */
     public static function checkIp4($requestIp, $ip)
     {
-        if (!filter_var($requestIp, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4)) {
-            return false;
-        }
+        $cacheKey = $requestIp.'-'.$ip;
+        if (!array_key_exists($cacheKey, self::$checkedIps)) {
```

Comment:
I would return early:

```php
if (array_key_exists($cacheKey, self::$checkedIps)) {
    return self::$checkedIps[$cacheKey];
}
```

I would even change `arra_key_exists()` to `isset()`